index.js

```diff
@@ -16,58 +14,153 @@ const serializeError = FJS({
   }
 })
 
-function rateLimitPlugin (fastify, opts, next) {
-  const timeWindow = typeof opts.timeWindow === 'string'
-    ? ms(opts.timeWindow)
-    : typeof opts.timeWindow === 'number'
-      ? opts.timeWindow
-      : 1000 * 60
-
-  const store = opts.redis
-    ? new RedisStore(opts.redis, timeWindow)
-    : new LocalStore(timeWindow, opts.cache)
+function buildRouteRate (pluginComponent, params, routeOptions, next) {
+  const urlT = (routeOptions.url === '/') ? 'root' : routeOptions.url.replace(/\//g, ':').slice(1)
 
-  const keyGenerator = typeof opts.keyGenerator === 'function'
-    ? opts.keyGenerator
-    : (req) => req.raw.ip
-
-  const skipOnError = opts.skipOnError === true
-  const max = opts.max || 1000
-  const whitelist = opts.whitelist || []
-  const after = ms(timeWindow, { long: true })
+  const prefix = {
+    cache: params.prefixCache ? params.prefixCache.replace(/[-,.;#*]/g, ':') : urlT
+  }
+  if (pluginComponent.isRedis && pluginComponent.whiteListInRedis) {
+    prefix.wl = urlT
```

Comment:
What's `urlT`? What's `wl`? Please use long, descriptive variable names.

Reply:
```
const urlT = (routeOptions.url === '/') ? 'root' : routeOptions.url.replace(/\//g, ':').slice(1)
```

`urlT` is the url transformed in case the user did not give a `prefixCache`. It will also make sure that if it's `/` ( root ), it will replace it with a string as by default I replace the `/` by `:`. 

`wl` is for the `whiteList`, it's a shortname for the variable. 

When using `redis` the key look like this by default : 

**For the whiteList define in the plugin (`global`)  **
````
${pluginComponent.app}:wlg:${prefix.wl}
````

**For the whiteList define in the `route` using { config : rateLimit : { whitelist : [...] } }**
````
${pluginComponent.app}:wle:${prefix.wl}
````

`pluginComponent.app` is the name of the app define in the `plugin` with the option `appName`. It's useful for clarity in redis.

Let me know if further question.

Reply:
Can you use log variables, and add the above explanation as a comment?

Reply:
Sure, Let me comment the whole code if you want. it will be easier to rewiew.

Reply:
Hello @mcollina , commented all the code. Should be better to read.